examples/ShoppingAds/AddListingScope.php

```diff
@@ -0,0 +1,184 @@
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param int $campaignId the campaign ID
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $campaignId
+    ) {
+
+        // A listing scope allows you to filter the products that will be included in a given
+        // campaign. You can specify multiple dimensions with conditions that must be met for a
+        // product to be included in a campaign.
+        // A typical ListingScope might only have a few dimensions. This example demonstrates a
+        // range of different dimensions you could use.
+        $listingScopeInfo = new ListingScopeInfo([
+            'dimensions' => [
+                // Creates a listing brand info set to "google".
+                new ListingDimensionInfo([
+                    'listing_brand'
+                    => new ListingBrandInfo(['value' => new StringValue(['value' => 'google'])])
+                ]),
+                // Creates a listing custom attribute info for INDEX0 set to "top_selling_products".
+                new ListingDimensionInfo([
+                    'listing_custom_attribute' => new ListingCustomAttributeInfo([
+                        'index' => ListingCustomAttributeIndex::INDEX0,
+                        'value' => new StringValue(['value' => 'top_selling_products'])
+                    ])
+                ]),
+                // Creates a product type info for LEVEL1 set to "electronics".
+                new ListingDimensionInfo([
+                    'product_type' => new ProductTypeInfo([
+                        'level' => ProductTypeLevel::LEVEL1,
+                        'value' => new StringValue(['value' => 'electronics'])
+                    ])
+                ]),
+                // Creates a product type info for LEVEL2 set to "smartphones".
+                new ListingDimensionInfo([
+                    'product_type' => new ProductTypeInfo([
+                        'level' => ProductTypeLevel::LEVEL2,
+                        'value' => new StringValue(['value' => 'smartphones'])
+                    ])
+                ])
+            ]
+        ]);
+
+        // Creates a campaign criterion to store the listing scope.
+        $campaignCriterion = new CampaignCriterion([
+            'campaign' =>
+                new StringValue(['value' => ResourceNames::forCampaign($customerId, $campaignId)]),
+            'listing_scope' => $listingScopeInfo
+        ]);
+
+        // Issues a mutate request to create a campaign criterion on the server and print its info.
+        $campaignCriterionServiceClient = $googleAdsClient->getCampaignCriterionServiceClient();
+        $response = $campaignCriterionServiceClient->mutateCampaignCriteria(
+            $customerId,
+            [new CampaignCriterionOperation(['create' => $campaignCriterion])]
```

Comment:
In our other code examples, the operation is created and stored in a dedicated variable. You might want to do the same here for consistency.

Reply:
Actually, this is more consistent with all other places, i.e., when creating an object, we now specify a field directly (like creating ListingDimensionInfo above).
We don't use `setProductType()` method, for instance.

Because of the above reason and its conciseness, I actually prefer this style.
But I just don't feel necessary to change all the examples to follow this.

I'll change this to match other examples for now, but please also let me know what you think in the long run. :)

Reply:
I prefer this one too: easily understandable, readable and more concise.